spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkV2Filters.java

```diff
@@ -289,11 +311,28 @@ private static Object convertLiteral(Literal<?> literal) {
     return literal.value();
   }
 
-  private static Expression handleEqual(String attribute, Object value) {
-    if (NaNUtil.isNaN(value)) {
-      return isNaN(attribute);
+  private static UnboundPredicate<Object> handleEqual(Pair<UnboundTerm<Object>, Object> children) {
+    if (children.second() == null) {
+      return isNull(children.first());
+    }
+
+    if (NaNUtil.isNaN(children.second())) {
+      return isNaN(children.first());
+    } else {
+      return equal(children.first(), children.second());
+    }
+  }
+
+  private static UnboundPredicate<Object> handleNotEqual(
```

Comment:
Sorry for not mentioning earlier, but can we make these two methods take not a Pair, but actual arguments, so its easier to read?

Then just invoke in the caller: , handle[Not]Equal(children.first(), children.second())